src/components/Exchange/PositionSeller.js

```diff
@@ -270,19 +340,52 @@ export default function PositionSeller(props) {
       }
     }
 
+    receiveToken = (allowReceiveTokenChange && swapToken) 
+      ? swapToken
+      : collateralToken;
+
     receiveAmount = receiveAmount.add(collateralDelta);
+    receiveAmount = receiveAmount.mul(bigNumberify(1000000));
+    convertedReceiveAmount = getTokenAmount(receiveAmount, receiveToken.address, false, infoTokens);
+
+    if (allowReceiveTokenChange && swapToken) {
+      const swapTokenInfo = getTokenInfo(infoTokens, swapToken.address);
+
+      notEnoughReceiveTokenLiquidity = swapTokenInfo.availableAmount.lt(convertedReceiveAmount);
+
+      const { feeBasisPoints } = getNextToAmount(
+        chainId,
+        fromAmount,
+        collateralToken.address,
+        receiveToken.address,
+        infoTokens,
+        undefined,
+        undefined,
+        usdgSupply,
+        totalTokenWeights,
+        true
+      );
+
+      if (feeBasisPoints) {
+        const baseFromAmountUsd = fromAmount.mul(BASIS_POINTS_DIVISOR).div(position.leverage)
```

Comment:
position.leverage might fluctuate based on pnl https://github.com/gmx-io/gmx-interface/blob/master/src/views/Exchange/Exchange.js#L299

user can enable a flag "include pnl in leverage calculation" in settings. i guess in this case calculation may be incorrect?